src/steps/do-query/do-get.js

```diff
@@ -63,20 +64,56 @@ export default function(requestContext, responseContext, registry) {
 
 }
 
-function parseFields(fieldsParam) {
+function parseFields(fieldsParam, registry) {
   let fields;
   if(typeof fieldsParam === "object") {
     fields = {};
     let isField = (it) => !arrayContains(["id", "type"], it);
 
     for(let type in fieldsParam) {
-      let provided = parseCommaSeparatedParam(fieldsParam[type]) || [];
+      let provided = parseCommaSeparatedParam(fieldsParam[type], type, registry) || [];
       fields[type] = provided.filter(isField);
     }
   }
   return fields;
 }
 
-function parseCommaSeparatedParam(it) {
-  return it ? it.split(",").map(decodeURIComponent) : undefined;
+function parseCommaSeparatedParam(it, type, registry) {
+  return it ? it.split(",").map(f => {
+    return transformDotSeparatedPath(f, type, registry);
+  }) : undefined;
+}
+
+function transformDotSeparatedPath(it, type, registry) {
+  let parts = it.split(".");
+  let types = [type];
+  parts.forEach((p, index) => {
+    if (index < parts.length - 1) {
+      types.push(registry.dbAdapter(types[index]).constructor.getReferencedType(types[index], p));
+    }
+  });
+
+  return parts.map((part, index) => {
+    if (shouldCamelizeType(types[index], registry)) {
+      return camelize(decodeURIComponent(part));
+    }
+    return decodeURIComponent(part);
+  }).join(".");
+}
+
+function transformFilters(it, type, registry) {
```

Comment:
> What if we retrieved an exhaustive list of field names for the resource from the adapter, and only ever transformed those? Can that be done reliably with Mongoose, accounting for virtuals and relationships and such?

That definitely can be done (it's what the documentation does), but given that the mongo directives are currently unaffected, let's not bother with this much complexity for now. We can do it "the right way" when the filter system is refactored per #15. Until then, lets just do the simplest thing that could possibly work, which is leaving it as is, so we can start using dasherizing in production!